sdk/monitor/azure-monitor-ingestion/azure/monitor/ingestion/_operations/_patch.py

```diff
@@ -52,6 +54,16 @@ def upload(  # pylint: disable=arguments-renamed, arguments-differ
         :rtype: None
         :raises: ~azure.core.exceptions.HttpResponseError
         """
+        if isinstance(logs, IOBase):
+            # Check if the stream is gzip-compressed..
+            content_encoding = None
+            if logs.read(2) == GZIP_MAGIC_NUMBER:
+                content_encoding = "gzip"
+            logs.seek(0)
```

Comment:
IOBase is not guaranteed to be seekable, right?

Reply:
Ah yea, that's right after looking into it. Good catch. I will refactor with some checks.

Reply:
Changed the logic to check for readable streams, and only do the gzip check if the stream is seekable. In most cases where an IO type is passed in, I believe the stream will be seekable.